pyomo/contrib/gdpopt/config_options.py

```diff
@@ -528,3 +528,39 @@ def _add_tolerance_configs(CONFIG):
             description="Tolerance for bound convergence.",
         ),
     )
+
+
+def _add_ldsda_configs(CONFIG):
+    CONFIG.declare(
+        "direction_norm",
+        ConfigValue(
+            default='L2',
+            domain=In(['L2', 'Linf']),
+            description="The norm to use for the search direction",
+        ),
+    )
+    CONFIG.declare(
+        "starting_point",
+        ConfigValue(default=None, description="The value list of external variables."),
+    )
+    CONFIG.declare(
+        "logical_constraint_list",
+        ConfigValue(
+            default=None,
+            description="""
+            The list of logical constraints to be reformulated into external variables.
```

Comment:
You are unclear here that you are anticipating that the input be names. However (see below), I think you should accept a list of components, and probably validate it here so that you don't have to later. Same for `disjunction_list`.

Reply:
The input must be a list of components from the `original_model`, but the algorithm operates on the `working_model`. We need to establish a link between them, ideally using component names, if components are provided. I'm open to improving this approach if there are better solutions.

Reply:
I have changed both `disjunction_list` and `logical_constraint_list` to lists of components instead of component names.